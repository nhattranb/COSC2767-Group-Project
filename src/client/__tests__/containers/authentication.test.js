/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 25 Dec 2024
 *   Last modified: 25 Dec 2024
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Authentication from '../../app/containers/Authentication';

const mockStore = configureStore([]);

describe('Authentication HOC', () => {
  let store;
  
  const DummyComponent = () => <div>Protected Component</div>;
  const WrappedComponent = Authentication(DummyComponent);

  beforeEach(() => {
    store = mockStore({
      authentication: {
        authenticated: false
      }
    });
  });

  it('redirects to login when not authenticated', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <WrappedComponent />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders protected component when authenticated', () => {
    store = mockStore({
      authentication: {
        authenticated: true
      }
    });

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <WrappedComponent />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 
