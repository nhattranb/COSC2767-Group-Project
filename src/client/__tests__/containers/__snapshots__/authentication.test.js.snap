// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication HOC redirects to login when not authenticated 1`] = `<DocumentFragment />`;

exports[`Authentication HOC renders protected component when authenticated 1`] = `
<DocumentFragment>
  <div>
    Protected Component
  </div>
</DocumentFragment>
`;
